### PR TITLE
Rubocop: address Rails/SaveBang exclusions

### PR DESCRIPTION
**Pull request by colinbruce**
## What

This addresses _most_ of the save_bang infractions

Unless we are checking the result of the call, we should use `.save!` rather than `.save` 
This will raise an error and is preferred

This PR leaves two of the uses that affect a lot of other classes.  
They can be addresses in separate PRs as it's likely changing `draftable.rb` could break tests in _all_ the request specs

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by agoldstone93 — APPROVED** [review]
One question but otherwise looks good

**Comment by agoldstone93** [comment]
Sorry, just realised this is still draft so you might not have been done. Whoops

File: features/step_definitions/civil_journey_steps.rb [hunk 1]
```diff
@@ -1090,10 +1090,12 @@
   fill_in(fields[2][:name].to_s, with: date.year)
 end
 
+# rubocop:disable Rails/SaveBang
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply features/step_definitions/civil_journey_steps.rb:1093]: I'm happy to leave it long term, I believe it's a false positive.  
This is a  Rails cop and is supposed looking for Rails.CRUD calls and changing to their bang versions.  This is a FactoryBot.create call and doesn't have a create! variant.

**Comment by colinbruce** [comment]
> Sorry, just realised this is still draft so you might not have been done. Whoops

No worries, the tests were still running when I logged off.  If it was successful I would have opened it for review this morning! 😄 

I'm also going to run some UAT regression tests as it is not just changing spec files :+1: so it will be hanging around a bit longer than some of my rubocop PRs

**Comment by colinbruce** [comment]
Successfully run through some applications and seen no issues
I'll merge and monitor

**Pull request merged**